buildenv/jenkins/JenkinsfileBase

```diff
@@ -108,34 +115,53 @@ def setupEnv() {
 def setupParallelEnv() {
 	stage('setupParallelEnv') {
 		timestamps{
-			setupEnv()
+			setupEnv()		
+			
+			//Initialization for things in common between both multiple iterations and many subtests
+			//repeatAmount is the number of times the loops go through
+			int repeatAmount = 0
 			def testSubDirs = []
-			def testSubDirSize = 0
+	
+			repeatAmount = ITERATIONS
 			
-			//For parallel suppose BUILD_LIST is one specific category
-			dir("$WORKSPACE/openjdk-tests/${env.BUILD_LIST}") {
-				testSubDirs = sh(returnStdout: true, script: "ls -d */").trim().tokenize()
-				testSubDirSize = testSubDirs.size()
+			//Setup for option-specific parameters
+			if (repeatAmount == 1){
+				//For parallel suppose BUILD_LIST is one specific category
+				dir("$WORKSPACE/openjdk-tests/${env.BUILD_LIST}") 
+				{
+					testSubDirs = sh(returnStdout: true, script: "ls -d */").trim().tokenize()
+					testSubDirSize = testSubDirs.size()
```

Comment:
- `testSubDirSize` is not defined anywhere. Actually, we do not need it. We can do

```
repeatAmount = testSubDirs.size()
``` 
- Please keep the original echo msg